app/src/main/java/com/example/concordia_campus_guide/Models/Routes/Route.java

```diff
@@ -0,0 +1,102 @@
+package com.example.concordia_campus_guide.Models.Routes;
+
+import com.example.concordia_campus_guide.ClassConstants;
+import java.util.ArrayList;
+import java.util.List;
+
+public class Route {
+
+    private List<TransportType> steps;
+    private String departureTime;
+    private String arrivalTime;
+    private String duration;
+    private String summary;
+    private @ClassConstants.TransportType String mainTransportType;
+
+    // For test
+    public Route() {
+        this.steps = new ArrayList<>();
+    }
+
+    // When TRANSIT is chosen
+    public Route(String departureTime, String arrivalTime, String duration, String mainTransportType) {
+        this.steps = new ArrayList<>();
+        this.departureTime = departureTime;
+        this.arrivalTime = arrivalTime;
+        this.duration = duration;
+        this.summary = "";
+        this.mainTransportType = mainTransportType;
+    }
+
+    // When TRANSIT is chosen but there's no arrivalTime and departureTime associated to it
+    public Route(String duration, String mainTransportType) {
+        this.steps = new ArrayList<>();
+        this.departureTime = "";
+        this.arrivalTime = "";
+        this.duration = duration;
+        this.summary = "";
```

Comment:
remind me a lot of builder design pattern

Reply:
yeah thought of that too, will look into it

Reply:
Due to lack of time, I'll open an improvement task for it. I won't be implementing the Builder pattern now.

Reply:
Issue #179 has been created